components/mediation-registry/org.wso2.carbon.mediation.registry/src/main/java/org/wso2/carbon/mediation/registry/MicroIntegratorRegistry.java

```diff
@@ -206,6 +208,97 @@ public OMNode lookup(String key) {
         return result;
     }
 
+    /**
+     * The micro integrator expects the properties of a directory to be available inside the given directory as a
+     * property file. For an example, if a directory key, conf:/foo/bar is passed as the key, the micro integrator
+     * registry expects the properties to be available in the file, conf:/foo/bar/bar.properties.
+     *
+     * @param key the path to the directory
+     * @return the properties defined
+     */
+    public Properties lookupProperties(String key) {
```

Comment:
Can't we use a one try catch block here? or throw exception and wrap the method inside a try catch block? WDYT?

Reply:
done